src/actions/xdrViewer.js

```diff
@@ -102,17 +100,21 @@ export function fetchSigners(input, horizonBaseUrl, networkPassphrase) {
           sigObj.isValid = isValid ? SIGNATURE.VALID : SIGNATURE.INVALID;
         }
         dispatch({
-          type: FETCHED_SIGNERS_SUCCESS,
+          type: FETCHED_SIGNERS.SUCCESS,
           result: signatures,
         });
       })
       .catch(e => {
         console.error(e);
-        dispatch({ type: FETCHED_SIGNERS_FAIL })
+        if (e.status == 404) {
+          dispatch({ type: FETCHED_SIGNERS.NOT_EXIST });
+        } else {
+          dispatch({ type: FETCHED_SIGNERS.FAIL });
+        }
       });
     } catch(e) {
       console.error(e);
-      dispatch({ type: FETCHED_SIGNERS_FAIL });
+      dispatch({ type: FETCHED_SIGNERS.FAIL });
```

Comment:
I am wondering if we might be missing something here by considering only signers that are fetched successfully.  This is because a signer can still be valid even if the account is not active on the network. @bartekn thoughts?

Reply:
When the signer does not exist, can we still verify the signature?

Reply:
Please note that we are only loading source accounts that must exist for transaction to succeed. It's true that you can use signer that does not have corresponding account in the ledger.